src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -255,11 +286,126 @@ Result SpatialJoinAlgorithms::BaselineAlgorithm() {
+  double withinDist = -1;
+  if (joinTypeVal == SpatialJoinType::WITHIN_DIST) {
+    withinDist = maxDist.value_or(0);
+    spatialJoin_.value()->runtimeInfo().addDetail("within-dist", withinDist);
+  }
+
+  // Configure the sweeper.
+  const sj::SweeperCfg sweeperCfg = [&] {
+    sj::SweeperCfg cfg;
+    cfg.numThreads = NUM_THREADS;
+    cfg.numCacheThreads = NUM_THREADS;
+    cfg.geomCacheMaxSize = 10000;
+    cfg.pairStart = "";
+    cfg.sepIsect = std::string{static_cast<char>(SpatialJoinType::INTERSECTS)};
+    cfg.sepContains = std::string{static_cast<char>(SpatialJoinType::CONTAINS)};
+    cfg.sepCovers = std::string{static_cast<char>(SpatialJoinType::COVERS)};
+    cfg.sepTouches = std::string{static_cast<char>(SpatialJoinType::TOUCHES)};
+    cfg.sepEquals = std::string{static_cast<char>(SpatialJoinType::EQUALS)};
+    cfg.sepOverlaps = std::string{static_cast<char>(SpatialJoinType::OVERLAPS)};
+    cfg.sepCrosses = std::string{static_cast<char>(SpatialJoinType::CROSSES)};
+    cfg.pairEnd = "";
+    cfg.useBoxIds = true;
+    cfg.useArea = true;
+    cfg.useOBB = false;
+    cfg.useCutouts = true;
+    cfg.useDiagBox = true;
+    cfg.useFastSweepSkip = true;
+    cfg.useInnerOuter = false;
+    cfg.noGeometryChecks = false;
+    cfg.withinDist = withinDist;
+    cfg.writeRelCb = [&results, &resultDists, joinTypeVal](
+                         size_t t, const char* a, const char* b,
+                         const char* pred) {
+      if (joinTypeVal == SpatialJoinType::WITHIN_DIST) {
+        results[t].push_back({std::atoi(a), std::atoi(b)});
+        resultDists[t].push_back(atof(pred));
+      } else if (pred[0] == static_cast<char>(joinTypeVal)) {
+        results[t].push_back({std::atoi(a), std::atoi(b)});
+      }
+    };
+    cfg.logCb = {};
+    cfg.statsCb = {};
+    cfg.sweepProgressCb = {};
+    return cfg;
+  }();
+
+  sj::Sweeper sweeper(sweeperCfg, ".", "", "spatialjoin");
+
+  ad_utility::Timer tParse{ad_utility::Timer::Started};
+
+  // Parse the geometries from the left and right input table, starting with the
+  // smaller one. Compute the bounding box of the smaller table (appropriately
+  // inflated for `WITHIN_DIST` joins) and only add those geometries from the
+  // larger table that intersect this bounding box.
+  if (idTableLeft->size() < idTableRight->size()) {
+    auto box = libspatialjoinParse(false, idTableLeft, leftJoinCol, sweeper,
+                                   NUM_THREADS);
+    sweeper.setFilterBox(box);
+    libspatialjoinParse(true, idTableRight, rightJoinCol, sweeper, NUM_THREADS);
+  } else {
+    auto box = libspatialjoinParse(true, idTableRight, rightJoinCol, sweeper,
+                                   NUM_THREADS);
+    sweeper.setFilterBox(box);
+    libspatialjoinParse(false, idTableLeft, leftJoinCol, sweeper, NUM_THREADS);
+  }
+
+  // Flush the geometries and add the time for parsing and processing the
+  // geometries to the runtime information.
+  sweeper.flush();
```

Comment:
Please explain more concretely (not verbosely) in the preceding comment in which was the geometries are "flushed" here. Are we just talking about flushing something to disk here, or is it also flushing objects from worker threads. Or yet something else?

Reply:
I rephrased this comment, `flush()` flushes the geometry caches and the sweep event list cache to disk.